vumi_twilio_api/tests/test_twilio_api.py

```diff
@@ -347,27 +384,161 @@ def test_make_call_optional_parameters_ifmachine(self):
             TwilioRestException)
         self.assertEqual(e.status, 400)
         message = json.loads(e.msg)
-        self.assertEqual(message['error_type'], 'UsageError')
+        self.assertEqual(message['error_type'], 'TwilioAPIUsageException')
         self.assertEqual(
             message['error_message'],
             "IfMachine value must be one of [None, 'Continue', 'Hangup']")
 
+    @inlineCallbacks
+    def test_make_call_ack_fallback_url(self):
+        self.twiml_server.add_err('err.xml', 'Error response')
+        response = twiml.Response()
+        self.twiml_server.add_response('default.xml', response)
+        yield self._twilio_client_create_call(
+            'err.xml', from_='+12345', to='+54321',
+            fallback_url='default.xml')
+        [msg] = yield self.app_helper.wait_for_dispatched_outbound(1)
+        yield self.app_helper.dispatch_event(self.app_helper.make_ack(msg))
+        [bad, req] = self.twiml_server.requests
+        self.assertEqual(req['filename'], 'default.xml')
+        self.assertEqual(bad['filename'], 'err.xml')
+
+    @inlineCallbacks
+    def test_make_call_ack_response(self):
+        response = twiml.Response()
+        self.twiml_server.add_response('default.xml', response)
+        yield self._twilio_client_create_call(
+            'default.xml', from_='+12345', to='+54321')
+        [msg] = yield self.app_helper.wait_for_dispatched_outbound(1)
+        yield self.app_helper.dispatch_event(self.app_helper.make_ack(msg))
+        [req] = self.twiml_server.requests
+        self.assertEqual(req['filename'], 'default.xml')
+        self.assertEqual(req['request'].args['CallStatus'], ['in-progress'])
+
+    @inlineCallbacks
+    def test_make_call_nack_response(self):
+        response = twiml.Response()
+        self.twiml_server.add_response('default.xml', response)
+        yield self._twilio_client_create_call(
+            'default.xml', from_='+12345', to='+54321')
+        [msg] = yield self.app_helper.wait_for_dispatched_outbound(1)
+        yield self.app_helper.dispatch_event(self.app_helper.make_nack(msg))
+        [req] = self.twiml_server.requests
+        self.assertEqual(req['filename'], 'default.xml')
+        self.assertEqual(req['request'].args['CallStatus'], ['failed'])
+
+    @inlineCallbacks
+    def test_make_call_parsing_twiml(self):
+        response = twiml.Response()
+        response.say('foobar')
+        self.twiml_server.add_response('default.xml', response)
+
+        twimls = []
+
+        def parse_Say(twiml):
+            twimls.append(twiml)
+        self.worker.twiml_parser._parse_Say = parse_Say
```

Comment:
What I'm trying to test is that this for loop is reached and run for each verb: https://github.com/praekelt/vumi-twilio-api/pull/5/files#diff-cfa655271fcdcea43cd84327ca643b3aR125 . Although I'm not sure what the best way is to test this. Suggestions?